apps/aehttp/src/aehttp_dispatch_ext.erl

```diff
@@ -163,6 +163,20 @@ handle_request('GetKeyBlockByHeight', Params, _Context) ->
             {404, [], #{reason => <<"Block not found">>}}
     end;
 
+handle_request('PostKeyBlock', #{'KeyBlock' := Data}, _Context) ->
```

Comment:
To me this seems like an attack vector. We have tried to avoid HTTP spamming by using Noise as protocol to exchange blocks. It cost to, probably little, but something to validate these blocks. In particular invalid blocks take a lot of CPU cycles (due to throwing exceptions deep down in the code being rather expensive).

We need a discussion on whether this really is a good idea.

Reply:
There used to be the same endpoint which was removed. Now, it's added back as it will allow creation of mining pools. The mining pool manager creates a key block (but doesn't mine) and the miners can request those via `/key-blocks/pending` (`pow` and `nonce` are not set). Miners then mine the key block, once a block is mined, they can submit the block by posting it to `/key-blocks`. Reward distribution and different mining strategies are to be developed by the community.

Reply:
Miners only solve the puzzle delivered by the pool manger,  if the hash meets the pool difficulty, they get a share,  if the hash also meets the AE network difficulty, the block has been successfully mined, in this case the pool submits the block to the AE network and everyone with shares gets paid accordingly. This article is great: [Bitcoin mining the hard way](http://www.righto.com/2014/02/bitcoin-mining-hard-way-algorithms.html).

If I'm right, only the pool manager request `/key-blocks[/pending]`, there is no need for miners to request this API?